internal/controller/storageclient_controller.go

```diff
@@ -421,9 +423,51 @@ func (r *storageClientReconcile) reconcilePhases() (ctrl.Result, error) {
 		}
 	}
 
+	//Once the owner references are removed from resources, StorageClaim is deleted
+	if err = r.deleteStorageClaims(); err != nil {
+		return reconcile.Result{}, err
+	}
 	return reconcile.Result{}, nil
 }
 
+func (r *storageClientReconcile) deleteStorageClaims() error {
+	storageClaimFinalizer := "storageclaim.ocs.openshift.io"
+	expectedStorageClaimNames := []string{
+		fmt.Sprintf("%s-ceph-rbd", r.storageClient.Name),
+		fmt.Sprintf("%s-cephfs", r.storageClient.Name),
+	}
+
+	for _, claimName := range expectedStorageClaimNames {
```

Comment:
We have 2 very specific claims? Why do we need a list and a loop to clean these?
The strightforward approach is the just have a `deleteStorageClaim` function and call it twice directly from `reconcilePhases`. Something like

```
if err := deleteStorageClaim(fmt.Sprintf("%s-ceph-rbd", r.storageClient.Name)); err != nil {

}
if err